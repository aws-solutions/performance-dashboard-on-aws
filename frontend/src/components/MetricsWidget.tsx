import React from "react";
import { Metric } from "../models";
import MetricsCardGroup from "../components/MetricsCardGroup";

type Props = {
  title: string;
  metrics: Array<Metric>;
  metricPerRow: number;
};

const MetricsWidget = (props: Props) => {
  const { title, metrics, metricPerRow } = props;

  return (
    <div>
      <h2 className="margin-top-4">{title}</h2>
      {metrics.length ? (
        <div>
          <MetricsCardGroup metrics={metrics} metricPerRow={metricPerRow} />
        </div>
      ) : (
        ""
      )}
    </div>
  );
};

export default MetricsWidget;
