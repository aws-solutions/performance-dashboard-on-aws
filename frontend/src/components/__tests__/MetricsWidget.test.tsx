/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import MetricsWidget from "../MetricsWidget";

test("renders the title and summary of the metrics preview component", async () => {
    const { getByText } = render(
        <MetricsWidget
            title="test title"
            metrics={[]}
            metricPerRow={3}
            significantDigitLabels={true}
            metricsCenterAlign={true}
        />,
        { wrapper: MemoryRouter },
    );
    expect(getByText("test title")).toBeInTheDocument();
});

test("metrics preview should match snapshot", async () => {
    const wrapper = render(
        <MetricsWidget
            title="test title"
            metrics={[{ title: "test title", value: 1 }]}
            metricPerRow={3}
            significantDigitLabels={true}
            metricsCenterAlign={true}
        />,
        { wrapper: MemoryRouter },
    );
    expect(wrapper.container).toMatchSnapshot();
});
