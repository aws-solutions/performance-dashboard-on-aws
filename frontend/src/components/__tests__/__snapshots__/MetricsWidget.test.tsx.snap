// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metrics preview should match snapshot 1`] = `
<div>
  <div>
    <h2
      class="margin-top-4"
    >
      test title
    </h2>
    <div>
      <div
        class="grid-col"
      >
        <div
          class="grid-row flex-column"
        >
          <div
            class="grid-col grid-row flex-row"
          >
            <div
              class="grid-col-4 padding-05"
            >
              <div
                class="display-flex flex-column border-base-lightest border-2px height-card padding-1"
              >
                <div
                  class="flex-5"
                >
                  <p
                    class="text-base-darkest text-bold margin-0"
                  >
                    test title
                  </p>
                </div>
                <div
                  class="flex-3 usa-tooltip"
                  data-position="bottom"
                  title="1"
                >
                  <h1
                    class="margin-0 text-no-wrap overflow-hidden text-overflow-ellipsis"
                  >
                    1
                  </h1>
                </div>
                <div
                  class="flex-2"
                />
                <div
                  class="flex-2"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
